src/InternalUtils.vala

```diff
@@ -348,5 +348,25 @@ namespace Gala {
 
             return actor_box;
         }
+
+        public delegate void WindowActorReadyCallback (Meta.WindowActor window_actor);
+
+        public static void wait_for_window_actor (Meta.Window window, owned WindowActorReadyCallback callback) {
+            unowned var window_actor = (Meta.WindowActor) window.get_compositor_private ();
+            if (window_actor != null) {
+                callback (window_actor);
+                return;
+            }
+
+            Idle.add (() => {
+                window_actor = (Meta.WindowActor) window.get_compositor_private ();
+
+                if (window_actor != null) {
+                    callback (window_actor);
+                }
+
+                return Source.REMOVE;
+            });
```

Comment:
g_object_ref is called on window when it's handed to the idle so there shouldn't be a way for it to be freed before that runs. 
Apart from that I think a null check wouldn't help because that only checks whether the pointer is null not whether the object has already been freed (I don't know a whole lot about c) and since we don't access anything in window directly we would be caught by get_compositor_private () and probably generate a warning instead of crash? Though it could very well be I'm wrong here somewhere since like I said I don't know a whole lot about c :sweat_smile: 
Still if you feel better I can add it, it won't hurt I guess :shrug:

Reply:
I don't know a lot about how all of this works under the hood neither so that's fine by me. Thanks for explaining